rfcs/80-curve-trait.md

```diff
@@ -0,0 +1,793 @@
+/// `interpolation(&x, &y, 1.0)` are equivalent to `x` and `y` respectively.
+pub fn new(
+    timed_samples: impl Into<Vec<(f32, T)>>,
+    interpolation: I,
+) -> Result<Self, UnevenSampleCurveError> { //... }
+```
+
+This kind of construction can be useful for strategically saving space when performing serialization and storage (although, keep in mind
+that the interpolating function may need to be separately handled in user-space). The `UnevenSampleCurve` type also supports 
+*forward* mapping of its sample times via the method `map_sample_times`, which accepts a function like the inverse of the one that 
+would be used in `Curve::reparametrize`.
+
+The associated resampling method is `resample_uneven`:
+```rust
+/// Resample this [`Curve`] to produce a new one that is defined by interpolation over samples
+/// taken at the given set of times. The given `sample_times` are expected to contain at least
+/// two valid times within the curve's domain range.
+///
+/// Irredundant sample times, non-finite sample times, and sample times outside of the domain
+/// are simply filtered out. With an insufficient quantity of data, a [`ResamplingError`] is
+/// returned.
+///
+/// The domain of the produced [`UnevenSampleCurve`] stretches between the first and last
+/// sample times of the iterator.
+fn resample_uneven(
+    &self,
+    sample_times: impl IntoIterator<Item = f32>,
+) -> Result<UnevenSampleCurve<T>, ResamplingError> { //... }
+```
+
+Note that there is no equivalent to `samples` for uneven spacing; however, these can be obtained easily using `Iterator::map`:
+```rust
+for (t, v) in (0..100).map(|x| x * x).map(|t| (t, my_curve.sample(t))) {
+    println!("Value {v} at time {t}.");
+}
+```
+The reason for this shortfall is that user expectations surrounding filtering, sorting, clamping, and so on are hard to predict. For example, 
+`sample` here could be replaced with `sample_clamped`, or the second `map` with `filter_map` and `sample` with `sample_checked`, among many other 
+variations. With time, perhaps a good default will become apparent and be added to the API.
+
+Finally, for some common math types whose interpolation is especially obvious and well-behaved (and enshrined in a trait), there are
+convenience methods `resample_auto` and `resample_uneven_auto` that use this type-inferred interpolation automatically. The expectation is not
+that interpolation in user-space will commonly use this trait and the associated methods, since the bar for having a valid trait implementation
+is rather high; rather, in domains where weaker interpolation notions are prevalent (e.g. animation), the expectation is that consumers will
+primarily go through `SampleCurve` or `UnevenSampleCurve` or define their own curve constructions entirely.
+
+### Combining curves
+
+There are a couple of common ways of combining curves that are supported by the Curve API. The first of these is `compose`, which appends two curves together end-to-end:
```

Comment:
I think it should be called `chain` rather than `compose`. Compose has a very vague meaning meaning, and chain is the name of the same operations on iterators in rust

Reply:
Good idea! I never really thought `compose` was a good name — I just wanted to make sure the functionality was included in some fashion :)